portfolio/src/main/webapp/index.html

```diff
@@ -23,14 +23,26 @@ <h1>Tal's Portfolio</h1>
       <p>Click here to get a random fact about me!:</p>
       <button id = "button" onclick="addRandomFact()">About Me</button>
       <div id="fact-container"></div>
-      <p>Click here to see a random greeting:</p>
-      <button onclick="getRandomGreeting()">Greet Me!</button>
-      <div id="quote-container"></div>
-
+ 
+      <p>Post a comment here!</p>
       <form method="POST" action="/data">
-        <input type="text" name="title" />
-          <br/>
+        <input type="text" name="text" />
+        <br/>
+        <button id = "button">Submit Comment</button>
+
       </form>
+    
+      <p>How many comments should display:</p>
+      <form method="POST" action="/data">
```

Comment:
This might be what's causing some of the issues you were having (including the one with undefined/blank comments).  In your servlet you have the "get an X number of comments" hooked up to your GET action, and you have your "post a new comment" action hooked up to your POST action.  Since this form here submits a POST request to the servlet, it would instead try to post a new comment, which would have been blank in this scenario since you don't have the boxes to fill in the comment.

Definitely try switching this to GET for starters.

Then maybe a good opportunity for improvement would be to add validations to your POST comment action.  eg. don't save into datastore if the comment text is missing, or even better if it returns an error and the page displays the error.